src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -399,47 +445,52 @@ void RNN::fillCellDesc() {
     }
     outDataDescs.reserve(S + 1);
 
-    inDataDescs.emplace_back(std::make_shared<DnnlBlockedMemoryDesc>(inShape, dataType, memory::format_tag::tnc));
-    outDataDescs.emplace_back(std::make_shared<DnnlBlockedMemoryDesc>(outShape, dataType, memory::format_tag::tnc));
+    // @todo use indexies instead of emplacing back, since order matters
+    inDataDescs.emplace_back(std::make_shared<DnnlBlockedMemoryDesc>(inShape, inDataTypes[xIdx], memory::format_tag::tnc));
+    outDataDescs.emplace_back(std::make_shared<DnnlBlockedMemoryDesc>(outShape, outDataTypes[yIdx], memory::format_tag::tnc));
```

Comment:
This change would lead to an exception with DIEN static model - like "[ ERROR ] could not create a descriptor for a GRU forward propagation primitive" in case of BF16.

As https://oneapi-src.github.io/oneDNN/dev_guide_rnn.html#implementation-details and https://github.com/oneapi-src/oneDNN/blob/f4cc3122a05762a1390a2407374da90bcf157a39/src/common/rnn.cpp#L115 indicts, both src_iter and dst_iter should be bf16 to support BF16 rnn.
In DIEN model, when enforecebf16 is used, the rnn node port 0 is of bf16, while port1 is still of f32. Then problem happens.

Can we revert this piece of code back to let dst_iter and src_iter use the same data type of port0?

Reply:
Fixed in subsequent PR